src/firewall/core/fw_zone.py

```diff
@@ -94,9 +94,16 @@ def get_zone_of_source(self, source):
                 return zone
         return None
 
-    def get_zone(self, zone):
-        z = self._fw.check_zone(zone)
-        return self._zones[z]
+    def check_zone(self, zone):
+        return self.get_zone(zone).name
+
+    def get_zone(self, zone, required=True):
```

Comment:
I'm still reading the patch series... but I don't follow this `required` argument. Why would we ever do `required=False`? I think the assumption in the caller already knows the zone exists.

Reply:
it's used all over the place.

in particular, there are many callers that want to check whether a zone name is valid. They call `get_zone(zone_name, required=False)`, to not be bothered by the not-found-exception.

Reply:
there are only a few callers that set `required=False`. Mostly because the previous users were replaced by `has_{zone,policy,ipset}()`, which is a wrapper around `get_{zone,policy,ipset}(name, required=False)`.

The remaining users that use `required=False` do need it as they are called at places where we are not sure that the object exists. Otherwise, they would have to inconveniently deal with an exception.